src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlValidator.cs

```diff
@@ -76,15 +76,14 @@ public static partial class Base64Url
             public static bool ValidateAndDecodeLength(int length, int paddingCount, out int decodedLength)
             {
                 // Padding is optional for Base64Url, so need to account remainder. If remainder is 1, then it's invalid.
-                int remainder = (int)((uint)length % 4);
-
+                (int whole, int remainder) = int.DivRem(length, 4);
                 if (remainder == 1)
                 {
                     decodedLength = 0;
                     return false;
                 }
 
-                decodedLength = (int)((uint)length / 4 * 3) + (remainder > 0 ? remainder - 1 : 0) - paddingCount;
+                decodedLength = (whole * 3) + (remainder > 0 ? remainder - 1 : 0) - paddingCount;
```

Comment:
```suggestion
                decodedLength = (int)((whole * 3) + (remainder > 0 ? remainder - 1 : 0) - paddingCount);
```
See [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgITxlRoG8aG+GADlACWAN2wYYDSADtcGBsJkKAstgTD8AV3wBRGZAAmMADIwZAcwwALBgF5FygHRqEANWwAbLVID0DFAYAKgY6AG5efki+eiZyJAZgCAhPBg9PYUMJGABBGUMAERgjU3MrawAKJQVPMps0RwUBbENDJQsAYQgtZQbuhWqGYxLDM0sbAEpohh5qfnmGX38ABRa2y0VcBggBDGEIGS8GADNoVnZUAFUoTwbcCAYZGBhDBgwH7DBIHoVYfGwlMYoE4GABJY4MP4A/IwKCbBjkBo2cyKDAAci2SnEGUMTmm8yqygYAHdrCkYA1BlDAbCJvZGk4CmIAEowfAVWrjawNFATCJzBZ8YQQirUmFwuwOchTAWC2aCwXDCDGUZ1WwOAAM/IVC2IAHYTl5cDBtQqAL40fH8JUqsblekVUnk4KhOkAagYorZ0KBDAAfAwNQwAPyQ700uFwBEMECBulR5qtdpdH6mwX6t5QHxphgW2UMaaxYjxRLJVLpTLZPKFYrK0pc8iEmpqylExPrTrdXrbLQDIk2l52ybTeUKpYMVZJjbCLY7PYHI6nOFsY1XG53B5PF5vD5fLu/cPikHgsP/CPwxFvawo4TozEybGZPH5glaQZO2oNN9EsVAukOb8MEZFk2QqCpAImTlyh5PkrT4ODFBFX9YXsKUZR1UcdT4AdVS5ektQQ+YM2OI0TQQvMEJwod1U9aoJjAj8pBCOh3U9ZC4QDINQ3Yhgo3IGM414wQ1mTfdYJffgMwwLMyPzPMzSAA) to spot the difference when using `uint` in these cases.

Reply:
That is so big diff, thank you!